src/pexprs-madlibs.js

```diff
@@ -0,0 +1,209 @@
+'use strict';
+
+// --------------------------------------------------------------------
+// Imports
+// --------------------------------------------------------------------
+
+var common = require('./common');
+var pexprs = require('./pexprs');
+
+// --------------------------------------------------------------------
+// Operations
+// --------------------------------------------------------------------
+
+pexprs.PExpr.prototype.generateExample = common.abstract;
+
+pexprs.any.generateExample = function(grammar, examples, inSyntacticContext, actuals) {
+  return {example: String.fromCharCode(Math.floor(Math.random() * 255))};
+};
+
+function categorizeExamples(examples) {
+  var examplesNeeded = examples.filter(function(example) {
+                                  return example.hasOwnProperty('examplesNeeded');
+                                })
+                               .map(function(example) { return example.examplesNeeded; });
+  examplesNeeded = [].concat.apply([], examplesNeeded)
```

Comment:
[].concat.apply([], list of lists) flattens the list of lists into a single list. Would it be more clear if I made this into an explicit 'flatten' function?